pkg/generate/git/repository.go

```diff
@@ -378,8 +379,7 @@ func timedCommand(timeout time.Duration, name, dir string, env []string, args ..
 
 	// check whether git was available in the first place
 	if err != nil {
-		_, err := exec.LookPath(name)
-		if err != nil {
+		if !isBinaryInstalled(name) {
 			return "", "", ErrGitNotAvailable
```

Comment:
umm, it's in the git package... As a reader of the code, I would assume that all functions here would be related to git. We have 2 constructors for the Repository object, one where you specify the binary and one where you don't. The gitserver which is also a client of this code allows you to specify the location of your git binary.